src/Microsoft.AspNet.Mvc.Core/DefaultControllerFactory.cs

```diff
@@ -2,27 +2,53 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Microsoft.AspNet.Http;
 using Microsoft.AspNet.Mvc.Core;
+using Microsoft.AspNet.Mvc.ModelBinding;
 using Microsoft.Framework.DependencyInjection;
 
 namespace Microsoft.AspNet.Mvc
 {
+    /// <summary>
+    /// Default implementation for <see cref="IControllerFactory"/>.
+    /// </summary>
     public class DefaultControllerFactory : IControllerFactory
     {
-        private readonly IServiceProvider _serviceProvider;
-        private readonly ITypeActivator _typeActivator;
         private readonly IControllerActivator _controllerActivator;
+        private readonly IDictionary<Type, Func<ActionContext, object>> _valueAccessorLookup;
+        private readonly ConcurrentDictionary<Type, PropertyActivator<ActionContext>[]> _activateActions;
+        private readonly Func<Type, PropertyActivator<ActionContext>[]> _getPropertiesToActivate;
 
-        public DefaultControllerFactory(IServiceProvider serviceProvider,
-                                        ITypeActivator typeActivator,
-                                        IControllerActivator controllerActivator)
+        /// <summary>
+        /// Initializes a new instance of <see cref="DefaultControllerFactory"/>.
+        /// </summary>
+        /// <param name="serviceProvider">Request scoped <see cref="IServiceProvider"/>.</param>
+        /// <param name="typeActivator"><see cref="ITypeActivator"/> used to instantiate types.</param>
+        /// <param name="controllerActivator"><see cref="IControllerActivator"/> used to activate instantiated
+        /// controllers.</param>
+        public DefaultControllerFactory(IControllerActivator controllerActivator)
         {
-            _serviceProvider = serviceProvider;
-            _typeActivator = typeActivator;
             _controllerActivator = controllerActivator;
+            _valueAccessorLookup = CreateValueAccessorLookup();
+            _activateActions = new ConcurrentDictionary<Type, PropertyActivator<ActionContext>[]>();
+
+            _getPropertiesToActivate = type => Enumerable.Concat(
+                PropertyActivator<ActionContext>.GetPropertiesToActivate(
+                                                        type,
+                                                        typeof(ActivateAttribute),
+                                                        CreateActivateInfo),
+                PropertyActivator<ActionContext>.GetPropertiesToActivate(
+                                                        type,
+                                                        typeof(FromServicesAttribute),
+                                                        CreateFromServicesInfo)).ToArray();
```

Comment:
I guess. That said, I'm guessing this is interim code and should go away once we do true model binding here.